### PR TITLE
Add support for Proxmox 

### LINKED ISSUE #1257: Add Support for creating Proxmox VM templates.
**Is your feature request related to a problem? Please describe.**

We are using Proxmox to provision vanilla Kubernetes clusters.
and we also implementing a capi provider.

Although, we can use qemu provider to build images
But, I think automating the creation of proxmox templates is a way better.

**Describe the solution you'd like**
Having a Proxmox target that enables to create VM, boot from ISO, install stuff. and then stop it, and convert it to template.

**Describe alternatives you've considered**

 packer proxmox builders can be used: https://developer.hashicorp.com/packer/plugins/builders/proxmox

build images manually with qemu.

**Additional context**
This can be achived via communicating with Proxmox API.

/kind feature

### PR DESCRIPTION
**Pull request by mcbenjemaa**
What this PR does / why we need it:

This PR will add a target for building Proxmox templates.
* ADD Support for Proxmox build

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged): Fixes #

Fixes #1257 

**Additional context**
Add any other context for the reviewers

### CODE REVIEW

**Review by AverageMarcus — COMMENTED** [review]
Looks good so far! 🎉 

@mcbenjemaa would you mind adding docs pages for the new provider too please?

**Comment by mcbenjemaa** [comment]
I'm back from vacation, i will rebase this.

**Comment by mcbenjemaa** [comment]
/cc @AverageMarcus @mboersma 

I'm ready

File: images/capi/Makefile [hunk 1]
```diff
@@ -193,6 +193,15 @@ deps-hcloud:
 	$(PACKER) init packer/config.pkr.hcl
 	$(PACKER) init packer/hcloud/config.pkr.hcl
 
+.PHONY: deps-proxmox
+deps-proxmox: ## Installs/checks dependencies for Proxmox builds
+deps-proxmox:
+	hack/ensure-ansible.sh
+	hack/ensure-packer.sh
```
Comment (AverageMarcus) [inline images/capi/Makefile:200]:
```suggestion
deps-proxmox: deps-common
```

Comment (AverageMarcus) [reply images/capi/Makefile:200]: Just this one change and then it's good to go :) 

We [recently made a change](https://github.com/kubernetes-sigs/image-builder/commit/a299f46009691d65a8b0505c85e92084d107b320) to group the common deps into a single target to ensure they don't get missed.

Comment (mcbenjemaa) [reply images/capi/Makefile:200]: ah yeah, I see

**Comment by AverageMarcus** [comment]
@mboersma  Do you happen to know why this test might be failing? It looks like an environment issue but I'm not sure.

@mcbenjemaa For now do you mind squashing your commits please?

**Pull request merged**